src/api/tp.js

```diff
@@ -67,19 +68,34 @@ var TargetProcess = (function() {
     }
 
     var me = this;
-    me.getTask(id).then(success, function (err) {
-        if(err.statusCode === 404) {
-            me.getBug(id).then(success, function (err) {
-                if(err.statusCode === 404) {
-                    done('Task/Bug with Id '+id+' could not be found or access is forbidden.');
-                }
-                else failure(err);
-            });
-        }
-        else failure(err);
+    me.getStory(id).then(function(result) {
+      if (!me.allowLoggingTimeToUserStories) {
```

Comment:
Given a TP id, we don't yet know whether this is an id for a Task/User Story/Bug.
Attempting to get it as a story with the Id means that OneTime can give the user feedback on why they couldn't use the story to register time against (because their config says they can't), rather than just reporting that it could not be found.

However, I have rejigged the order, so we only attempt to get a User Story after failing to get a Task and a Bug.